falcon/util/uri.py

```diff
@@ -554,7 +564,6 @@ def unquote_string(quoted: str) -> str:
 
 # TODO(vytas): Restructure this in favour of a cleaner way to hoist the pure
 # Cython functions into this module.
-if not TYPE_CHECKING:  # pragma: nocover
```

Comment:
Does it hurt to keep `TYPE_CHECKING` here? The idea was not even to mess with cythonized replacements, if we are type checking.

Reply:
no, but I don't really see a need. the type checker is not complaining, so I don't see why we should bother with `if not TYPE_CHECKING`